interfaces.go

```diff
@@ -5,14 +5,31 @@
 package optimize
 
 // A Method can optimize an objective function.
+//
+// It uses a reverse-communication interface design represented by the Init()
+// and Iterate() methods to pass and retrieve information from the client. Its
```

Comment:
It does when MajorIteration is returned. In that case the whole Location might be filled by Method.

I am going to rewrite this paragraph anyway.